README.md

```diff
@@ -227,6 +228,88 @@ module "terraform-aws-basic-layout" {
     tags = local.tags
 }
 ```
+#### ec2-default-instance-profile-ssm
+
+```terraform
+module "terraform-aws-basic-layout" {
+    source = "../../"
+
+    prefix                      = var.prefix
+    name                        = var.name
+
+    aws_ami_os_id               = var.aws_ami_os_id
+    aws_ami_os_owner            = var.aws_ami_os_owner
+    instance_type               = var.instance_type
+    vpc_id                      = data.terraform_remote_state.vpc.outputs.vpc_id
+    subnet_id                   = data.terraform_remote_state.vpc.outputs.public_subnets[0]
+    associate_public_ip_address = var.associate_public_ip_address
+    key_pair_name               = data.terraform_remote_state.security.outputs.aws_key_pair_name
+    ebs_optimized               = var.ebs_optimized
+    monitoring                  = var.monitoring
+    user_data_base64            = base64encode(local.user_data)
+    enable_ssm_access           = local.enable_ssm_access
+
+    root_block_device = [
+        {
+            volume_type = "gp2"
+            volume_size = 10
+            encrypted   = true
+        },
+    ]
+
+    ebs_block_device = [
+        {
+            device_name = "/dev/sdf"
+            volume_type = "gp2"
+            volume_size = 5
+            encrypted   = true
+        },
+        {
+            device_name = "/dev/sdg"
+            volume_type = "gp2"
+            volume_size = 5
+            encrypted   = true
+        }
+    ]
+
+    security_group_rules = [
+        {
+            from_port   = 8080,
+            to_port     = 8082,
+            protocol    = "tcp",
+            cidr_blocks = [ "0.0.0.0/0" ],
+            description = "Allow nginx proxy"
+        }
+    ]
+
+    dns_records_internal_hosted_zone = [{
+        zone_id = data.terraform_remote_state.vpc.outputs.aws_internal_zone_id[0],
+        name    = "ec2-basic-layout.aws.binbash.com.ar",
+        type    = "A",
+        ttl     = 300
+    }]
+
+    dns_records_public_hosted_zone = [{
+        zone_id = data.terraform_remote_state.vpc.outputs.aws_public_zone_id[0],
+        name    = "ec2-basic-layout.binbash.com.ar",
+        type    = "A",
+        ttl     = 300
+    }]
+
+    # EC2 IAM Profile (Role + Policies)
+    cross_account_roles_resource_arn_list = [
+        "arn:aws:iam::${var.dev_account_id}:role/DevOps",
+        "arn:aws:iam::${var.shared_account_id}:role/DevOps",
+        "arn:aws:iam::${var.dev_account_id}:role/Auditor",
+        "arn:aws:iam::${var.shared_account_id}:role/Auditor",
+    ]
+    policy_arn = [
+        "arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess",
+        "arn:aws:iam::aws:policy/CloudWatchReadOnlyAccess",
+    ]
+
+    tags = local.tags
+}
```

Comment:
@fgauchat-binbash for SSM what I imagined was some very simple steps to connect via SSM in the README.md, example set AWS credentials either via IAM or SSO and then run:

```
$ leverage aws sso configure
$ export AWS_SHARED_CREDENTIALS_FILE="/Users/exequielbarrirero/.aws/bb/credentials"
$ export AWS_CONFIG_FILE="/Users/exequielbarrirero/.aws/bb/config"
$ aws ssm start-session --target INSTANCE_ID
```

Or it  could even be a link to an external ref too if better.

And this example of the module would better fit at the code level, which is what we style, not necessarily at the root context README.md.

Probably it could be in the README.md of the particular example => https://github.com/binbashar/terraform-aws-ec2-basic-layout/tree/master/examples

- TF Registry Module |  https://registry.terraform.io/modules/binbashar/ec2-basic-layout/aws/latest
- TF Registry Example | Example https://registry.terraform.io/modules/binbashar/ec2-basic-layout/aws/latest/examples/ec2-default-instance-profile

As a result the /example/example-name/README.md it natively integrated and shown at the registry too. And as a good practice could be then used to automatically run terratests at CI time. In this case we could add the example code, but not necessary to add the associated tests (we should just leave them for a next round as agreed).

<img width="1470" alt="image" src="https://user-images.githubusercontent.com/18539704/170290966-58674681-0d8a-4d82-b57e-1ff9c5238c17.png">

Reply:
noted @exequielrafaela, I'll work on these changes and let you know